internal/verifier/migration_verifier.go

```diff
@@ -392,22 +397,27 @@ func (verifier *Verifier) getGenerationWhileLocked() (int, bool) {
 	return verifier.generation, verifier.lastGeneration
 }
 
+func (verifier *Verifier) maybeAppendGlobalFilter(predicates bson.A) bson.A {
```

Comment:
The name of the function is a bit confusing, Other readers might think  `globalFilter` is changed in this function.